test/Integration/Test/Executable.hs

```diff
@@ -95,7 +95,8 @@ testSmoke :: Assertion
 testSmoke = do
     ports@Ports { apiPort, ekgPort, metricsPort } <- randomPorts
     withOscoinExe ports $ \stdoutHdl _ -> do
-        out <- C8.hGet stdoutHdl 2048
+        threadDelay 2000000
```

Comment:
Any way we could avoid the artificial delay here? I can see how it might cause the tests to become slightly less deterministic. If it was inserted with the intent of account for boot time, shouldn't `hGet` block up until we have read the requested bytes or that is only the max buffer size and it might read actually less (too lazy to RTFM 😛 ). Alternatively, I guess we could try to loop over reads to `hGet` up until we don't read the requested number of bytes (which would require extending a little bit the tests to keep around a buffer of what we have read so far).

Reply:
Yeah idk. This started to fail at some point, but I couldn't be arsed to investigate. One would think that `hGet` blocks, but apparently it doesn't.